src/coreclr/jit/lsra.h

```diff
@@ -2466,16 +2466,25 @@ class RefPosition
     RefPosition* nextRefPosition;
 
     // The remaining fields are common to both options
-    GenTree*     treeNode;
+    union
+    {
+        struct
+        {
+            GenTree* treeNode;
+
+            // Prior to the allocation pass, registerAssignment captures the valid registers
+            // for this RefPosition.
+            // After the allocation pass, this contains the actual assignment
+            SingleTypeRegSet registerAssignment;
+        };
+#ifdef HAS_MORE_THAN_64_REGISTERS
+        regMaskTP killRegisterAssignment;
+#endif
+    };
```

Comment:
I think you told me why, but what was the reason why we cannot use the space provided by `referent` to make this require no additional space?

It might be worth to introduce a `class RefPositionKill : RefPosition` that has the extra field to avoid making `RefPosition` larger on arm64 (and x64 in the future).

Reply:
Ah, disregard this. I see that it already requires no additional space.

Reply:
It seems like it should be fine to make this union unconditional (i.e. without the `HAS_MORE_THAN_64_REGISTERS` define) to avoid a few ifdefs.